soql-server-pg/src/main/scala/com/socrata/pg/server/QueryServer.scala

```diff
@@ -116,60 +117,82 @@ class QueryServer(val dsInfo: DSInfo, val caseSensitivity: CaseSensitivity, val
-    val servReq = req.servletRequest
-    val copy = Option(servReq.getParameter("copy"))
-    val withFieldName = Option(servReq.getParameter("fieldName")).map(java.lang.Boolean.parseBoolean(_)).getOrElse(false)
-
-    withPgu(dsInfo, truthStoreDatasetInfo = None) { pgu =>
-      // get schema by dataset id or resource name
-      val copyInfoOpt = Option(servReq.getParameter("ds")) match {
-        case Some(ds) =>
-          getCopy(pgu, ds, copy)
-        case None => Option(servReq.getParameter("rn")) match {
-          case Some(rn) =>
-            getCopy(pgu, DatasetResourceName(rn), copy)
-          case None =>
-            None
+    def schema(req: HttpRequest): HttpResponse = {
+      val servReq = req.servletRequest
+      val copy = Option(servReq.getParameter("copy"))
+      val withFieldName = Option(servReq.getParameter("fieldName")).map(java.lang.Boolean.parseBoolean(_)).getOrElse(false)
+
+      withPgu(dsInfo, truthStoreDatasetInfo = None) { pgu =>
+        // get schema by dataset id or resource name
+        val copyInfoOpt = Option(servReq.getParameter("ds")) match {
+          case Some(ds) =>
+            getCopy(pgu, ds, copy)
+          case None => Option(servReq.getParameter("rn")) match {
+            case Some(rn) =>
+              getCopy(pgu, DatasetResourceName(rn), copy)
+            case None =>
+              None
+          }
         }
-      }
 
-      copyInfoOpt match {
-        case Some(copyInfo) =>
-          if (withFieldName) {
-            import com.socrata.pg.ExtendedSchema._
-            val schema =  getSchemaWithFieldName(pgu, copyInfo)
-            OK ~>
+        copyInfoOpt match {
+          case Some(copyInfo) =>
+            if (withFieldName) {
+              import com.socrata.pg.ExtendedSchema._
+              val schema =  getSchemaWithFieldName(pgu, copyInfo)
+              OK ~>
               copyInfoHeaderForSchema(copyInfo.copyNumber, copyInfo.dataVersion, copyInfo.lastModified) ~>
               Write(JsonContentType)(JsonUtil.writeJson(_, schema, buffer = true))
-          } else {
-            import com.socrata.pg.Schema._
-            val schema =  getSchema(pgu, copyInfo)
-            OK ~>
+            } else {
+              import com.socrata.pg.Schema._
+              val schema =  getSchema(pgu, copyInfo)
+              OK ~>
               copyInfoHeaderForSchema(copyInfo.copyNumber, copyInfo.dataVersion, copyInfo.lastModified) ~>
               Write(JsonContentType)(JsonUtil.writeJson(_, schema, buffer = true))
-          }
-        case None => NotFound
+            }
+          case None => NotFound
+        }
       }
     }
   }
 
-  object RollupResource extends SimpleResource {
-    override val get = rollups _
+  object ExistsResource extends SimpleResource {
+    override val get = exists _
+    def exists(req: HttpRequest): HttpResponse = {
+      val servReq = req.servletRequest
+
+      val dsOp = Option(servReq.getParameter("ds"))
```

Comment:
Making this optional is kinda weird but ok.

Reply:
oh that is weird. Fixed!